contract/models/contract.py

```diff
@@ -213,13 +213,73 @@ def action_contract_send(self):
         }
 
     @api.model
-    def _get_recurring_create_invoice_domain(self, contract=False):
+    def _finalize_invoice_values(self, invoice_values):
+        """
+        This method adds the missing values in the invoice lines dictionaries.
+
+        If no account on the product, the invoice lines account is
+        taken from the invoice's journal in _onchange_product_id
+        This code is not in finalize_creation_from_contract because it's
+        not possible to create an invoice line with no account
+
+        :param invoice_values: dictionary (invoice values)
+        :return: updated dictionary (invoice values)
+        """
+        # If no account on the product, the invoice lines account is
+        # taken from the invoice's journal in _onchange_product_id
+        # This code is not in finalize_creation_from_contract because it's
+        # not possible to create an invoice line with no account
+        new_invoice = self.env['account.invoice'].new(invoice_values)
+        for invoice_line in new_invoice.invoice_line_ids:
+            name = invoice_line.name
+            account_analytic_id = invoice_line.account_analytic_id
+            price_unit = invoice_line.price_unit
```

Comment:
@sbidoul @sbejaoui I'm not a fan of what I had to do here: we need to preserve `name`, `account_analytic_id` and `price_unit` because they come from the contract and they will be overwritten by the onchange (which is here to find the account from the invoice's journal if missing). A better idea?

Reply:
@ThomasBinsfeld Do we really need to call onchange on lines? For instance, when generating an invoice from a SO in standard Odoo, how is it done?

Reply:
@sbidoul They don't try to find it on the invoice's journal. If nothing found from the product -> error. https://github.com/odoo/odoo/blob/12.0/addons/sale/models/sale.py#L1330

Reply:
Hm, and there are the taxes to find too. In the SO they have it on the SO line, we don't have them on the contract line. I think your hack is good enough.